sql/handler.h

```diff
@@ -963,11 +971,12 @@ struct xid_recovery_member
   bool decided_to_commit;
   Binlog_offset binlog_coord; // semisync recovery binlog offset
   XID *full_xid;           // needed by wsrep or past it recovery
+  ulong server_id;         // server id of orginal server
 
   xid_recovery_member(my_xid xid_arg, uint prepare_arg, bool decided_arg,
-                      XID *full_xid_arg)
+                      XID *full_xid_arg, ulong server_id_arg)
```

Comment:
And the same `decltype(::server_id)` to replace `ulong` that you won't to explain why anymore.